### PR TITLE
tidy up and fix issue with handling enum values

### LINKED ISSUE #173: Issue with rendering single enum values
image

There is only one value for `type` (`tracking-event`) in this case. However, the app still displays it under `Valid values`, including the `...`. Clicking the `...` does not do anything.

### PR DESCRIPTION
**Pull request by quangkhoa**
Resolve #173

### CODE REVIEW

File: src/components/Property/Property.js [hunk 1]
```diff
@@ -13,18 +13,29 @@ export default class Property extends PureComponent {
...
+      if (enumValues.length === 1) {
+        name = `${name} = "${enumValues[0]}"`
+        enumValues = trimmedEnumValues = null
+      } else if (enumValues.length > 20) {
+        trimmedEnumValues = enumValues.slice(0, 20)
+        isEnumTrimmed = true
+      }
+    }
+
+    this.state = {
```
Comment (brendo) [inline src/components/Property/Property.js]: Should use `setState` I believe.

File: src/components/Property/Property.js [hunk 2]
```diff
@@ -41,12 +52,13 @@ export default class Property extends PureComponent {
...
 
-    let {name, enumValues} = this.props
+    const name = this.state.name
+    let enumValues
 
-    // If enumValues only has one single value, append the single value to name, and not display enum values
-    if (enumValues && enumValues.length === 1) {
-      name = `${name} = "${enumValues[0]}"`
-      enumValues = []
+    if (this.state.isEnumTrimmed) {
```
Comment (brendo) [inline src/components/Property/Property.js]: This prop is accessed twice, did you want to destructure `this.state` to get it?

**Review by brendo — CHANGES_REQUESTED** [review]

File: src/components/Property/Property.js (hunk 1 above)
Comment (quangkhoa) [reply src/components/Property/Property.js]: updated

File: src/components/Property/Property.js (hunk 2 above)
Comment (quangkhoa) [reply src/components/Property/Property.js]: Good pickup. Fixed.

**Review by brendo — APPROVED** [review]

**Pull request merged**